.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,38 @@
+# Contributing Guidelines for TerseTS
+
+The repository is a Zig project with C and Python bindings. It uses the Apache‑2.0 license and includes a pre‑commit hook for building, formatting, and running tests. The following guidelines summarize how to contribute effectively.
+
+## 1. Environment Setup 
+* Install a recent Zig toolchain (v0.14.0 is used in the GitHub workflow).
+* For Python bindings, install `setuptools` (and `build` for distribution) as shown in the Python README.
+* Build the project in `Debug` mode with `zig build` and in Release modes with `zig build -Doptimize=ReleaseFast`, etc., as outlined in the README.
+
+## 2. Running Checks
+* Use the provided tools/pre-commit script before committing. It builds the library, runs tests, checks formatting, and compiles the C header across Debug and ReleaseFast profiles.
```

Comment:
Maybe tell the reader that this script is intended to be used as a Git hook and either describe how to install it as such or point to documentation about how to do this.